src/Skia/Avalonia.Skia/FormattedTextImpl.cs

```diff
@@ -1,37 +1,42 @@
+using Avalonia.Media;
+using Avalonia.Platform;
+using SkiaSharp;
 using System;
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
 using System.Text;
-using Avalonia.Media;
-using Avalonia.Platform;
-using SkiaSharp;
 
 namespace Avalonia.Skia
 {
     unsafe class FormattedTextImpl : IFormattedTextImpl
     {
         public SKPaint Paint { get; private set; }
 
-        public FormattedTextImpl(string text)
+        public FormattedTextImpl(string text, TextWrapping wrapping = TextWrapping.NoWrap)
         {
             _text = text;
+            _wrapping = wrapping;
             Paint = new SKPaint();
 
+            //currently Skia does not measure properly with Utf8 !!!
+            //Paint.TextEncoding = SKTextEncoding.Utf8;
             Paint.TextEncoding = SKTextEncoding.Utf16;
             Paint.IsStroke = false;
             Paint.IsAntialias = true;
             LineOffset = 0;
 
             // Replace 0 characters with zero-width spaces (200B)
             _text = _text.Replace((char)0, (char)0x200B);
+
+            _text = _text.Replace("\r\n", "\r").Replace('\n','\r');
```

Comment:
You are changing the number of characters in the string. This will break HitTestTextPosition and HitTestTextRange